pkg/pxc/app/statefulset/haproxy.go

```diff
@@ -307,8 +307,9 @@ func (c *HAProxy) LogCollectorContainer(_ *api.LogCollectorSpec, _ string, _ str
 }
 
 func (c *HAProxy) PMMContainer(ctx context.Context, cl client.Client, spec *api.PMMSpec, secret *corev1.Secret, cr *api.PerconaXtraDBCluster) (*corev1.Container, error) {
-	if cr.CompareVersionWith("1.9.0") < 0 {
-		return nil, nil
+	clusterName := cr.Name
+	if cr.CompareVersionWith("1.18.0") >= 0 && cr.Spec.PMM.CustomClusterName != "" {
+		clusterName = cr.Spec.PMM.CustomClusterName
```

Comment:
Given that this behaviour is the same for all the PMM containers across the different pods, why don't we move this initialization and the respective env var to `app.PMMClient`?

Reply:
The same comment applies for all the components, meaning proxysql and node.